core/os/os.cpp

```diff
@@ -359,7 +362,46 @@ String OS::get_unique_id() const {
 }
 
 int OS::get_processor_count() const {
+#if defined(_WIN32) || defined(_WIN64)
+	DWORD length = 0;
+	int concurrency = 0;
+	const auto validConcurrency = [&concurrency]() noexcept -> int {
```

Comment:
We avoid lambdas [unless necessary](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#lambdas)